src/cmd/src/frontend.rs

```diff
@@ -136,14 +132,16 @@ impl StartCommand {
             opts.logging.level = level;
         }
 
-        let tls_option = TlsOption::new(
+        let tls_opts = TlsOption::new(
             self.tls_mode.clone(),
             self.tls_cert_path.clone(),
             self.tls_key_path.clone(),
         );
 
         if let Some(addr) = self.http_addr.clone() {
-            opts.http_options.get_or_insert_with(Default::default).addr = addr;
+            if let Some(http_opts) = &mut opts.http_options {
```

Comment:
Maybe we should use the original code. Seems they have the same behaviors when `opts.http_options` is Some.

Reply:
@v0y4g3r it doesn't change the original behavior. The [`load_layered_config()`](https://github.com/zyy17/greptimedb/blob/refactor/support-layered-configuration/src/cmd/src/options.rs#L67) will merge three kinds of configs(env+config_file+defaults). If the cli doesn't provide value, it will use the default value, which is `get_or_insert_with(Default::default)`. The default value of `FrontendOptions::default()` is always [not None](https://github.com/zyy17/greptimedb/blob/refactor/support-layered-configuration/src/frontend/src/frontend.rs#L49).